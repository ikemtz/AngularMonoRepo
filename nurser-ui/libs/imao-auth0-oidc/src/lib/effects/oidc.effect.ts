import { Inject, Injectable } from '@angular/core';
import { Actions, Effect, ofType } from '@ngrx/effects';
import { Action } from '@ngrx/store';
import { User as OidcUser } from 'oidc-client';
import { Observable, of } from 'rxjs';
import { catchError, concatMap, map, switchMap, tap } from 'rxjs/operators';
import { OidcActions } from '../actions';
import { Config, OIDC_CONFIG } from '../models/config.model';
import { ACTION_NO_ACTION } from '../models/constants';
import { OidcService } from '../services/oidc.service';

@Injectable()
export class OidcEffects {
  constructor(
    private actions$: Actions,
    @Inject(OidcService) private oidcService: OidcService,
    @Inject(OIDC_CONFIG) private config: Config
  ) {}

  @Effect()
  getOicdUser$ = this.actions$.pipe(
    ofType(OidcActions.OidcActionTypes.GetOidcUser),
    map((action: OidcActions.GetOidcUser) => action.payload),
    concatMap(args =>
      this.oidcService.getOidcUser().pipe(
        concatMap((userData: OidcUser) => {
          const r: Action[] = [new OidcActions.UserFound(userData)];
          const automaticSilentRenew =
            this.config.oidc_config.automaticSilentRenew != null && this.config.oidc_config.automaticSilentRenew;
          // user expired, initiate silent sign-in if configured to automatic
          if (userData != null && userData.expired === true && automaticSilentRenew === true) {
            r.push(new OidcActions.SigninSilent(args));
          }
          return r;
        }),
        catchError(error => {
          return of(new OidcActions.UserDoneLoading());
        })
      )
    )
  );
  @Effect()
  removeOidcUser$ = this.actions$.pipe(
    ofType(OidcActions.OidcActionTypes.RemoveOidcUser),
    concatMap(() => {
      return this.oidcService.removeOidcUser().pipe(
        concatMap(() => [new OidcActions.UserDoneLoading()]),
        catchError(error => [new OidcActions.OidcError(error)]) //
      );
    })
  );

  @Effect()
  userFound$ = this.actions$.pipe(
    ofType(OidcActions.OidcActionTypes.UserFound),
    concatMap(() => {
      return [new OidcActions.UserDoneLoading()];
    })
  );

  @Effect()
  onUserLoaded$: Observable<Action> = this.actions$.pipe(
    ofType(OidcActions.OidcActionTypes.OnUserLoaded),
    map((action: OidcActions.OnUserLoaded) => action.payload),
    switchMap((userData: OidcUser) => {
      return [new OidcActions.UserFound(userData)];
    })
  );

  @Effect()
  signInPopup$: Observable<Action> = this.actions$.pipe(
    ofType(OidcActions.OidcActionTypes.SignInPopup),
    map((action: OidcActions.SigninPopup) => action.payload),
    concatMap(args => {
      return this.oidcService.signInPopup(args).pipe(
        concatMap((user: OidcUser) => of({ type: ACTION_NO_ACTION })), // dispatch empty action
        catchError(error => of(new OidcActions.SignInError(error)))
      );
    })
  );

  @Effect()
  signInRedirect$: Observable<Action> = this.actions$.pipe(
    ofType(OidcActions.OidcActionTypes.SignInRedirect),
    map((action: OidcActions.SigninRedirect) => action.payload),
    concatMap(args => {
      return this.oidcService.signInRedirect(args).pipe(
        concatMap((user: OidcUser) => of({ type: ACTION_NO_ACTION })), // dispatch empty action
        catchError(error => of(new OidcActions.SignInError(error)))
      );
    })
  );

  @Effect()
  signInSilent$ = this.actions$.pipe(
    ofType(OidcActions.OidcActionTypes.SignInSilent),
    map((action: OidcActions.SigninSilent) => action.payload),
    concatMap(args => {
      return this.oidcService.signInSilent(args).pipe(
        concatMap((userData: OidcUser) => {
          return [new OidcActions.UserFound(userData)];
        }),
        catchError(error => {
          // Something went wrong renewing the access token.
          // Set loading done so the auth guard will resolve.
          return of(new OidcActions.OnSilentRenewError(error), new OidcActions.UserDoneLoading());
        })
      );
    })
  );

  @Effect()
  signOutPopup$: Observable<Action> = this.actions$.pipe(
    ofType(OidcActions.OidcActionTypes.SignOutPopup),
    map((action: OidcActions.SignoutPopup) => action.payload),
    concatMap(args => {
      return this.oidcService.signOutPopup(args).pipe(
        concatMap(() => of({ type: ACTION_NO_ACTION })), // dispatch empty action
        catchError(error => of(new OidcActions.SignOutError(error)))
      );
    })
  );

  @Effect()
  signOutRedirect$: Observable<Action> = this.actions$.pipe(
    ofType(OidcActions.OidcActionTypes.SignOutRedirect),
    map((action: OidcActions.SignoutRedirect) => action.payload),
    concatMap(args => {
      return this.oidcService.signOutRedirect(args).pipe(
        concatMap(() => of({ type: ACTION_NO_ACTION })), // dispatch empty action
        catchError(error => of(new OidcActions.SignOutError(error)))
      );
    })
  );
}
