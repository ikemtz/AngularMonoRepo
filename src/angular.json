{
  "$schema": "./node_modules/@angular/cli/lib/config/schema.json",
  "version": 1,
  "newProjectRoot": "",
  "projects": {
    "imng-kendo-odata": {
      "projectType": "library",
      "root": "libs/imng-kendo-odata",
      "sourceRoot": "libs/imng-kendo-odata/src",
      "prefix": "imng",
      "architect": {
        "build": {
          "builder": "@nrwl/angular:package",
          "options": {
            "tsConfig": "libs/imng-kendo-odata/tsconfig.lib.json",
            "project": "libs/imng-kendo-odata/ng-package.json"
          },
          "configurations": {
            "production": {
              "tsConfig": "libs/imng-kendo-odata/tsconfig.lib.prod.json"
            }
          }
        },
        "lint": {
          "builder": "@angular-devkit/build-angular:tslint",
          "options": {
            "tsConfig": ["libs/imng-kendo-odata/tsconfig.lib.json", "libs/imng-kendo-odata/tsconfig.spec.json"],
            "exclude": ["**/node_modules/**", "!libs/imng-kendo-odata/**"]
          }
        },
        "test": {
          "builder": "@nrwl/jest:jest",
          "options": {
            "jestConfig": "libs/imng-kendo-odata/jest.config.js",
            "tsConfig": "libs/imng-kendo-odata/tsconfig.spec.json",
            "passWithNoTests": true,
            "setupFile": "libs/imng-kendo-odata/src/test-setup.ts"
          }
        }
      },
      "schematics": {}
    },
    "imng-auth0-oidc": {
      "projectType": "library",
      "root": "libs/imng-auth0-oidc",
      "sourceRoot": "libs/imng-auth0-oidc/src",
      "architect": {
        "build": {
          "builder": "@nrwl/angular:package",
          "options": {
            "tsConfig": "libs/imng-auth0-oidc/tsconfig.lib.json",
            "project": "libs/imng-auth0-oidc/ng-package.json"
          },
          "configurations": {
            "production": {
              "tsConfig": "libs/imng-auth0-oidc/tsconfig.lib.prod.json"
            }
          }
        },
        "lint": {
          "builder": "@angular-devkit/build-angular:tslint",
          "options": {
            "tsConfig": ["libs/imng-auth0-oidc/tsconfig.lib.json", "libs/imng-auth0-oidc/tsconfig.spec.json"],
            "exclude": ["**/node_modules/**", "!libs/imng-auth0-oidc/**"]
          }
        },
        "test": {
          "builder": "@nrwl/jest:jest",
          "options": {
            "jestConfig": "libs/imng-auth0-oidc/jest.config.js",
            "tsConfig": "libs/imng-auth0-oidc/tsconfig.spec.json",
            "setupFile": "libs/imng-auth0-oidc/src/test-setup.ts",
            "passWithNoTests": true
          }
        }
      },
      "schematics": {
        "@nrwl/angular:component": {
          "styleext": "scss"
        }
      }
    },
    "imng-kendo-grid-odata": {
      "projectType": "library",
      "root": "libs/imng-kendo-grid-odata",
      "sourceRoot": "libs/imng-kendo-grid-odata/src",
      "architect": {
        "build": {
          "builder": "@nrwl/angular:package",
          "options": {
            "tsConfig": "libs/imng-kendo-grid-odata/tsconfig.lib.json",
            "project": "libs/imng-kendo-grid-odata/ng-package.json"
          },
          "configurations": {
            "production": {
              "tsConfig": "libs/imng-kendo-grid-odata/tsconfig.lib.prod.json"
            }
          }
        },
        "lint": {
          "builder": "@angular-devkit/build-angular:tslint",
          "options": {
            "tsConfig": [
              "libs/imng-kendo-grid-odata/tsconfig.lib.json",
              "libs/imng-kendo-grid-odata/tsconfig.spec.json"
            ],
            "exclude": ["**/node_modules/**", "!libs/imng-kendo-grid-odata/**"]
          }
        },
        "test": {
          "builder": "@nrwl/jest:jest",
          "options": {
            "jestConfig": "libs/imng-kendo-grid-odata/jest.config.js",
            "tsConfig": "libs/imng-kendo-grid-odata/tsconfig.spec.json",
            "setupFile": "libs/imng-kendo-grid-odata/src/test-setup.ts",
            "passWithNoTests": true
          }
        }
      },
      "schematics": {
        "@nrwl/angular:component": {
          "styleext": "scss"
        }
      }
    },
    "imng-application-insights-ngrx": {
      "projectType": "library",
      "root": "libs/imng-application-insights-ngrx",
      "sourceRoot": "libs/imng-application-insights-ngrx/src",
      "architect": {
        "build": {
          "builder": "@nrwl/angular:package",
          "options": {
            "tsConfig": "libs/imng-application-insights-ngrx/tsconfig.lib.json",
            "project": "libs/imng-application-insights-ngrx/ng-package.json"
          },
          "configurations": {
            "production": {
              "tsConfig": "libs/imng-application-insights-ngrx/tsconfig.lib.prod.json"
            }
          }
        },
        "lint": {
          "builder": "@angular-devkit/build-angular:tslint",
          "options": {
            "tsConfig": [
              "libs/imng-application-insights-ngrx/tsconfig.lib.json",
              "libs/imng-application-insights-ngrx/tsconfig.spec.json"
            ],
            "exclude": ["**/node_modules/**", "!libs/imng-application-insights-ngrx/**"]
          }
        },
        "test": {
          "builder": "@nrwl/jest:jest",
          "options": {
            "jestConfig": "libs/imng-application-insights-ngrx/jest.config.js",
            "tsConfig": "libs/imng-application-insights-ngrx/tsconfig.spec.json",
            "setupFile": "libs/imng-application-insights-ngrx/src/test-setup.ts",
            "passWithNoTests": true
          }
        }
      },
      "schematics": {
        "@nrwl/angular:component": {
          "styleext": "scss"
        }
      }
    },
    "imng-kendo-data-entry": {
      "projectType": "library",
      "root": "libs/imng-kendo-data-entry",
      "sourceRoot": "libs/imng-kendo-data-entry/src",
      "architect": {
        "build": {
          "builder": "@nrwl/angular:package",
          "options": {
            "tsConfig": "libs/imng-kendo-data-entry/tsconfig.lib.json",
            "project": "libs/imng-kendo-data-entry/ng-package.json"
          },
          "configurations": {
            "production": {
              "tsConfig": "libs/imng-kendo-data-entry/tsconfig.lib.prod.json"
            }
          }
        },
        "lint": {
          "builder": "@angular-devkit/build-angular:tslint",
          "options": {
            "tsConfig": [
              "libs/imng-kendo-data-entry/tsconfig.lib.json",
              "libs/imng-kendo-data-entry/tsconfig.spec.json"
            ],
            "exclude": ["**/node_modules/**", "!libs/imng-kendo-data-entry/**"]
          }
        },
        "test": {
          "builder": "@nrwl/jest:jest",
          "options": {
            "jestConfig": "libs/imng-kendo-data-entry/jest.config.js",
            "tsConfig": "libs/imng-kendo-data-entry/tsconfig.spec.json",
            "setupFile": "libs/imng-kendo-data-entry/src/test-setup.ts",
            "passWithNoTests": true
          }
        }
      },
      "schematics": {
        "@nrwl/angular:component": {
          "styleext": "scss"
        }
      }
    },
    "imng-ngrx-utils": {
      "projectType": "library",
      "root": "libs/imng-ngrx-utils",
      "sourceRoot": "libs/imng-ngrx-utils/src",
      "prefix": "imng",
      "architect": {
        "build": {
          "builder": "@nrwl/angular:package",
          "options": {
            "tsConfig": "libs/imng-ngrx-utils/tsconfig.lib.json",
            "project": "libs/imng-ngrx-utils/ng-package.json"
          },
          "configurations": {
            "production": {
              "tsConfig": "libs/imng-ngrx-utils/tsconfig.lib.prod.json"
            }
          }
        },
        "lint": {
          "builder": "@angular-devkit/build-angular:tslint",
          "options": {
            "tsConfig": ["libs/imng-ngrx-utils/tsconfig.lib.json", "libs/imng-ngrx-utils/tsconfig.spec.json"],
            "exclude": ["**/node_modules/**", "!libs/imng-ngrx-utils/**"]
          }
        },
        "test": {
          "builder": "@nrwl/jest:jest",
          "options": {
            "jestConfig": "libs/imng-ngrx-utils/jest.config.js",
            "tsConfig": "libs/imng-ngrx-utils/tsconfig.spec.json",
            "setupFile": "libs/imng-ngrx-utils/src/test-setup.ts",
            "passWithNoTests": true
          }
        }
      },
      "schematics": {
        "@nrwl/angular:component": {
          "styleext": "scss"
        }
      }
    },
    "imng-nrsrx-client-utils": {
      "projectType": "library",
      "root": "libs/imng-nrsrx-client-utils",
      "sourceRoot": "libs/imng-nrsrx-client-utils/src",
      "prefix": "imng",
      "architect": {
        "build": {
          "builder": "@nrwl/angular:package",
          "options": {
            "tsConfig": "libs/imng-nrsrx-client-utils/tsconfig.lib.json",
            "project": "libs/imng-nrsrx-client-utils/ng-package.json"
          },
          "configurations": {
            "production": {
              "tsConfig": "libs/imng-nrsrx-client-utils/tsconfig.lib.prod.json"
            }
          }
        },
        "lint": {
          "builder": "@angular-devkit/build-angular:tslint",
          "options": {
            "tsConfig": [
              "libs/imng-nrsrx-client-utils/tsconfig.lib.json",
              "libs/imng-nrsrx-client-utils/tsconfig.spec.json"
            ],
            "exclude": ["**/node_modules/**", "!libs/imng-nrsrx-client-utils/**"]
          }
        },
        "test": {
          "builder": "@nrwl/jest:jest",
          "options": {
            "jestConfig": "libs/imng-nrsrx-client-utils/jest.config.js",
            "tsConfig": "libs/imng-nrsrx-client-utils/tsconfig.spec.json",
            "setupFile": "libs/imng-nrsrx-client-utils/src/test-setup.ts",
            "passWithNoTests": true
          }
        }
      },
      "schematics": {
        "@nrwl/angular:component": {
          "styleext": "scss"
        }
      }
    },
    "imng-ngxb-typeahead": {
      "projectType": "library",
      "root": "libs/imng-ngxb-typeahead",
      "sourceRoot": "libs/imng-ngxb-typeahead/src",
      "architect": {
        "build": {
          "builder": "@nrwl/angular:package",
          "options": {
            "tsConfig": "libs/imng-ngxb-typeahead/tsconfig.lib.json",
            "project": "libs/imng-ngxb-typeahead/ng-package.json"
          },
          "configurations": {
            "production": {
              "tsConfig": "libs/imng-ngxb-typeahead/tsconfig.lib.prod.json"
            }
          }
        },
        "lint": {
          "builder": "@angular-devkit/build-angular:tslint",
          "options": {
            "tsConfig": ["libs/imng-ngxb-typeahead/tsconfig.lib.json", "libs/imng-ngxb-typeahead/tsconfig.spec.json"],
            "exclude": ["**/node_modules/**", "!libs/imng-ngxb-typeahead/**"]
          }
        },
        "test": {
          "builder": "@nrwl/jest:jest",
          "options": {
            "jestConfig": "libs/imng-ngxb-typeahead/jest.config.js",
            "tsConfig": "libs/imng-ngxb-typeahead/tsconfig.spec.json",
            "setupFile": "libs/imng-ngxb-typeahead/src/test-setup.ts",
            "passWithNoTests": true
          }
        }
      },
      "schematics": {
        "@nrwl/angular:component": {
          "styleext": "scss"
        }
      }
    },
    "imng-kendo-chart-odata": {
      "projectType": "library",
      "root": "libs/imng-kendo-chart-odata",
      "sourceRoot": "libs/imng-kendo-chart-odata/src",
      "architect": {
        "build": {
          "builder": "@nrwl/angular:package",
          "options": {
            "tsConfig": "libs/imng-kendo-chart-odata/tsconfig.lib.json",
            "project": "libs/imng-kendo-chart-odata/ng-package.json"
          },
          "configurations": {
            "production": {
              "tsConfig": "libs/imng-kendo-chart-odata/tsconfig.lib.prod.json"
            }
          }
        },
        "lint": {
          "builder": "@angular-devkit/build-angular:tslint",
          "options": {
            "tsConfig": [
              "libs/imng-kendo-chart-odata/tsconfig.lib.json",
              "libs/imng-kendo-chart-odata/tsconfig.spec.json"
            ],
            "exclude": ["**/node_modules/**", "!libs/imng-kendo-chart-odata/**"]
          }
        },
        "test": {
          "builder": "@nrwl/jest:jest",
          "options": {
            "jestConfig": "libs/imng-kendo-chart-odata/jest.config.js",
            "tsConfig": "libs/imng-kendo-chart-odata/tsconfig.spec.json",
            "setupFile": "libs/imng-kendo-chart-odata/src/test-setup.ts",
            "passWithNoTests": true
          }
        }
      },
      "schematics": {
        "@nrwl/angular:component": {
          "styleext": "scss"
        }
      }
    },
    "imng-az-func-configr": {
      "projectType": "application",
      "schematics": {
        "@nrwl/angular:component": {
          "style": "scss"
        }
      },
      "root": "apps/imng-az-func-configr",
      "sourceRoot": "apps/imng-az-func-configr/src",
      "prefix": "imng",
      "architect": {
        "build": {
          "builder": "@angular-devkit/build-angular:browser",
          "options": {
            "outputPath": "dist/apps/imng-az-func-configr",
            "index": "apps/imng-az-func-configr/src/index.html",
            "main": "apps/imng-az-func-configr/src/main.ts",
            "polyfills": "apps/imng-az-func-configr/src/polyfills.ts",
            "tsConfig": "apps/imng-az-func-configr/tsconfig.app.json",
            "aot": true,
            "assets": [
              "apps/imng-az-func-configr/src/favicon.ico",
              "apps/imng-az-func-configr/src/assets",
              "apps/imng-az-func-configr/src/manifest.webmanifest"
            ],
            "styles": [
              "./node_modules/bootstrap/dist/css/bootstrap.min.css",
              "./node_modules/jsoneditor/dist/jsoneditor.min.css",
              "apps/imng-az-func-configr/src/styles.scss"
            ],
            "scripts": []
          },
          "configurations": {
            "production": {
              "fileReplacements": [
                {
                  "replace": "apps/imng-az-func-configr/src/environments/environment.ts",
                  "with": "apps/imng-az-func-configr/src/environments/environment.prod.ts"
                }
              ],
              "optimization": true,
              "outputHashing": "all",
              "sourceMap": false,
              "extractCss": true,
              "namedChunks": false,
              "extractLicenses": true,
              "vendorChunk": false,
              "buildOptimizer": true,
              "budgets": [
                {
                  "type": "initial",
                  "maximumWarning": "2mb",
                  "maximumError": "5mb"
                },
                {
                  "type": "anyComponentStyle",
                  "maximumWarning": "6kb",
                  "maximumError": "10kb"
                }
              ],
              "serviceWorker": true,
              "ngswConfigPath": "apps/imng-az-func-configr/ngsw-config.json"
            }
          }
        },
        "serve": {
          "builder": "@angular-devkit/build-angular:dev-server",
          "options": {
            "browserTarget": "imng-az-func-configr:build"
          },
          "configurations": {
            "production": {
              "browserTarget": "imng-az-func-configr:build:production"
            }
          }
        },
        "extract-i18n": {
          "builder": "@angular-devkit/build-angular:extract-i18n",
          "options": {
            "browserTarget": "imng-az-func-configr:build"
          }
        },
        "lint": {
          "builder": "@angular-devkit/build-angular:tslint",
          "options": {
            "tsConfig": ["apps/imng-az-func-configr/tsconfig.app.json", "apps/imng-az-func-configr/tsconfig.spec.json"],
            "exclude": ["**/node_modules/**", "!apps/imng-az-func-configr/**"]
          }
        },
        "test": {
          "builder": "@nrwl/jest:jest",
          "options": {
            "jestConfig": "apps/imng-az-func-configr/jest.config.js",
            "tsConfig": "apps/imng-az-func-configr/tsconfig.spec.json",
            "setupFile": "apps/imng-az-func-configr/src/test-setup.ts",
            "passWithNoTests": true
          }
        }
      }
    },
    "imng-az-func-configr-e2e": {
      "root": "apps/imng-az-func-configr-e2e",
      "sourceRoot": "apps/imng-az-func-configr-e2e/src",
      "projectType": "application",
      "architect": {
        "e2e": {
          "builder": "@nrwl/cypress:cypress",
          "options": {
            "cypressConfig": "apps/imng-az-func-configr-e2e/cypress.json",
            "tsConfig": "apps/imng-az-func-configr-e2e/tsconfig.e2e.json",
            "devServerTarget": "imng-az-func-configr:serve"
          },
          "configurations": {
            "production": {
              "devServerTarget": "imng-az-func-configr:serve:production"
            }
          }
        },
        "lint": {
          "builder": "@angular-devkit/build-angular:tslint",
          "options": {
            "tsConfig": ["apps/imng-az-func-configr-e2e/tsconfig.e2e.json"],
            "exclude": ["**/node_modules/**", "!apps/imng-az-func-configr-e2e/**"]
          }
        }
      }
    },
    "imng-snippets": {
      "projectType": "library",
      "root": "libs/imng-snippets",
      "sourceRoot": "libs/imng-snippets/src",
      "prefix": "imng",
      "architect": {
        "lint": {
          "builder": "@angular-devkit/build-angular:tslint",
          "options": {
            "tsConfig": ["libs/imng-snippets/tsconfig.lib.json", "libs/imng-snippets/tsconfig.spec.json"],
            "exclude": ["**/node_modules/**", "!libs/imng-snippets/**"]
          }
        },
        "test": {
          "builder": "@nrwl/jest:jest",
          "options": {
            "jestConfig": "libs/imng-snippets/jest.config.js",
            "tsConfig": "libs/imng-snippets/tsconfig.spec.json",
            "setupFile": "libs/imng-snippets/src/test-setup.ts",
            "passWithNoTests": true
          }
        }
      },
      "schematics": {}
    },
    "imng-kendo-grid": {
      "projectType": "library",
      "root": "libs/imng-kendo-grid",
      "sourceRoot": "libs/imng-kendo-grid/src",
      "prefix": "imng",
      "architect": {
        "build": {
          "builder": "@nrwl/angular:package",
          "options": {
            "tsConfig": "libs/imng-kendo-grid/tsconfig.lib.json",
            "project": "libs/imng-kendo-grid/ng-package.json"
          },
          "configurations": {
            "production": {
              "tsConfig": "libs/imng-kendo-grid/tsconfig.lib.prod.json"
            }
          }
        },
        "lint": {
          "builder": "@angular-devkit/build-angular:tslint",
          "options": {
            "tsConfig": ["libs/imng-kendo-grid/tsconfig.lib.json", "libs/imng-kendo-grid/tsconfig.spec.json"],
            "exclude": ["**/node_modules/**", "!libs/imng-kendo-grid/**"]
          }
        },
        "test": {
          "builder": "@nrwl/jest:jest",
          "options": {
            "jestConfig": "libs/imng-kendo-grid/jest.config.js",
            "tsConfig": "libs/imng-kendo-grid/tsconfig.spec.json",
            "setupFile": "libs/imng-kendo-grid/src/test-setup.ts",
            "passWithNoTests": true
          }
        }
      },
      "schematics": {
        "@nrwl/angular:component": {
          "styleext": "scss"
        }
      }
    },
    "nurse-cron": {
      "projectType": "application",
      "schematics": {
        "@nrwl/angular:component": {
          "style": "scss"
        }
      },
      "root": "apps/nurse-cron",
      "sourceRoot": "apps/nurse-cron/src",
      "prefix": "nrcrn",
      "architect": {
        "build": {
          "builder": "@angular-devkit/build-angular:browser",
          "options": {
            "outputPath": "dist/apps/nurse-cron",
            "index": "apps/nurse-cron/src/index.html",
            "main": "apps/nurse-cron/src/main.ts",
            "polyfills": "apps/nurse-cron/src/polyfills.ts",
            "tsConfig": "apps/nurse-cron/tsconfig.app.json",
            "aot": true,
            "assets": [
              "apps/nurse-cron/src/favicon.ico",
              "apps/nurse-cron/src/assets",
              {
                "glob": "*",
                "input": "libs/imng-auth0-oidc/src/assets",
                "output": "/"
              }
            ],
            "styles": ["apps/nurse-cron/src/styles.scss"],
            "scripts": []
          },
          "configurations": {
            "production": {
              "fileReplacements": [
                {
                  "replace": "apps/nurse-cron/src/environments/environment.ts",
                  "with": "apps/nurse-cron/src/environments/environment.prod.ts"
                }
              ],
              "optimization": true,
              "outputHashing": "all",
              "sourceMap": false,
              "extractCss": true,
              "namedChunks": false,
              "extractLicenses": true,
              "vendorChunk": false,
              "buildOptimizer": true,
              "budgets": [
                {
                  "type": "initial",
                  "maximumWarning": "2mb",
                  "maximumError": "5mb"
                },
                {
                  "type": "anyComponentStyle",
                  "maximumWarning": "6kb",
                  "maximumError": "10kb"
                }
              ]
            }
          }
        },
        "serve": {
          "builder": "@angular-devkit/build-angular:dev-server",
          "options": {
            "browserTarget": "nurse-cron:build",
            "proxyConfig": "apps/nurse-cron/proxy.conf.json"
          },
          "configurations": {
            "production": {
              "browserTarget": "nurse-cron:build:production",
              "proxyConfig": "apps/nurse-cron/proxy.conf.json"
            }
          }
        },
        "extract-i18n": {
          "builder": "@angular-devkit/build-angular:extract-i18n",
          "options": {
            "browserTarget": "nurse-cron:build"
          }
        },
        "lint": {
          "builder": "@angular-devkit/build-angular:tslint",
          "options": {
            "tsConfig": ["apps/nurse-cron/tsconfig.app.json", "apps/nurse-cron/tsconfig.spec.json"],
            "exclude": ["**/node_modules/**", "!apps/nurse-cron/**"]
          }
        },
        "test": {
          "builder": "@nrwl/jest:jest",
          "options": {
            "jestConfig": "apps/nurse-cron/jest.config.js",
            "tsConfig": "apps/nurse-cron/tsconfig.spec.json",
            "passWithNoTests": true,
            "setupFile": "apps/nurse-cron/src/test-setup.ts"
          }
        }
      }
    },
    "nurse-cron-e2e": {
      "root": "apps/nurse-cron-e2e",
      "sourceRoot": "apps/nurse-cron-e2e/src",
      "projectType": "application",
      "architect": {
        "e2e": {
          "builder": "@nrwl/cypress:cypress",
          "options": {
            "cypressConfig": "apps/nurse-cron-e2e/cypress.json",
            "tsConfig": "apps/nurse-cron-e2e/tsconfig.e2e.json",
            "devServerTarget": "nurse-cron:serve"
          },
          "configurations": {
            "production": {
              "devServerTarget": "nurse-cron:serve:production"
            }
          }
        },
        "lint": {
          "builder": "@angular-devkit/build-angular:tslint",
          "options": {
            "tsConfig": ["apps/nurse-cron-e2e/tsconfig.e2e.json"],
            "exclude": ["**/node_modules/**", "apps/nurse-cron-e2e/src/plugins/index.js", "!apps/nurse-cron-e2e/**"]
          }
        }
      }
    }
  },
  "cli": {
    "warnings": {
      "typescriptMismatch": false,
      "versionMismatch": false
    },
    "defaultCollection": "@nrwl/angular",
    "analytics": "f4c251a3-a9e0-433e-b1f3-2c10661975ea"
  },
  "schematics": {
    "@nrwl/angular:application": {
      "unitTestRunner": "jest",
      "e2eTestRunner": "cypress"
    },
    "@nrwl/angular:library": {
      "unitTestRunner": "jest"
    }
  },
  "defaultProject": "nurse-cron"
}
