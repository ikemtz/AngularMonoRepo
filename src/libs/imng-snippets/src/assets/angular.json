{
  "ComponentMethodsTypeAhead": {
    "prefix": "imng-component-methods-type-ahead",
    "body": [
      "  public onTypeAheadSelected(match: ImngMatchSelectedEvent<I${1:__ENTITY_NAME__PASCAL_CASE__}>) {",
      "  }",
      "  public onTypeAheadCleared(noMatches: boolean) {",
      "  }"
    ],
    "description": "ComponentMethodsTypeAhead"
  },
  "ServicesDataEntry": {
    "prefix": "imng-services-data-entry",
    "body": [
      "import { HttpClient } from '@angular/common/http';",
      "import { Injectable } from '@angular/core';",
      "import { NrsrxBaseApiClientService } from 'imng-nrsrx-client-utils';",
      "@Injectable({",
      "  providedIn: 'root',",
      "})",
      "export class ${1:__ENTITY_NAME__PASCAL_CASE__}Service extends NrsrxBaseApiClientService<I${1:__ENTITY_NAME__PASCAL_CASE__}> {",
      "  url: string = environment.endPoints.${2:__NGRX_FEATURE_KEY__CAMEL_CASE__}s.${1:__ENTITY_NAME__PASCAL_CASE__};",
      "  constructor(http: HttpClient) {",
      "    super(http);",
      "  }",
      "}"
    ],
    "description": "ServicesDataEntry"
  }
}
