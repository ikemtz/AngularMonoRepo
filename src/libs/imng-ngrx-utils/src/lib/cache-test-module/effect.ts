import { Injectable, inject } from '@angular/core';
import { Store } from '@ngrx/store';
import * as actions from './actions';
import { map, switchMap } from 'rxjs/operators';

import { Actions, createEffect, ofType } from '@ngrx/effects';
import { handleEffectError } from '../error-handlers';
import { useCacheIfExists } from '../use-cache-if-exists';
import { feature } from './feature';
import { HttpClient } from '@angular/common/http';
import { DataRecord } from './data-record';

@Injectable()
export class Effects {
  private readonly actions$ = inject(Actions);
  private readonly store = inject(Store);
  private readonly httpClient = inject(HttpClient);

  loadEffect$ = createEffect(() => {
    return this.actions$.pipe(
      ofType(actions.loadRecordsRequest),
      useCacheIfExists(this.store, feature.selectRecords),
      switchMap((action: ReturnType<typeof actions.loadRecordsRequest>) =>
        this.httpClient.get<DataRecord[]>(`/api/cache-test`).pipe(
          map((t) => actions.loadRecordsSuccess(t)),
          handleEffectError(action),
        ),
      ),
    );
  });
}
