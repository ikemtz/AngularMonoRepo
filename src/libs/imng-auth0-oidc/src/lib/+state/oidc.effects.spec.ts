import { TestBed } from '@angular/core/testing';
import { Observable, of } from 'rxjs';

import { provideMockActions } from '@ngrx/effects/testing';
import { provideMockStore } from '@ngrx/store/testing';

import { NxModule, DataPersistence } from '@nrwl/angular';
import { hot, readFirst } from '@nrwl/angular/testing';
import { OidcEffects } from './oidc.effect';
import * as OidcActions from './oidc.action';
import { OIDC_CONFIG } from '../models/config.model';

describe('Oidc Effects', () => {
  let actions: Observable<any>;
  let effects: OidcEffects;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [NxModule.forRoot()],
      providers: [
        OidcEffects,
        DataPersistence,
        provideMockActions(() => actions),
        provideMockStore(),
        { provide: OIDC_CONFIG, useValue: { oidc_config: {} } }
      ],

    });

    effects = TestBed.inject(OidcEffects);
  });

  describe('signInSilent$', () => {
    it('should fail because silent_redirect_uri is not configured', async done => {
      try {
        actions = of(OidcActions.signinSilent({}));

        const result = await readFirst(effects.signInSilent$);
        expect(result).toMatchSnapshot();
        done();
      }
      catch (err) {
        done.fail(err);
      }
    });
  });
});
