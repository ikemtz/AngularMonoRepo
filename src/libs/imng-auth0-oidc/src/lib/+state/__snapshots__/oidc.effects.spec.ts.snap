// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Oidc Effects signInSilent$ should fail because silent_redirect_uri is not configured 1`] = `
Object {
  "payload": [Error: No silent_redirect_uri configured],
  "type": "[Oidc] on silent renew error",
}
`;
